File: swift/removeDuplicatesFromSortedArray.swift
func removeDuplicates(inout nums: [Int]) -> Int {
    if nums.count > 0 {
        for i in 0..<nums.count {
            while i + 1 < nums.count && nums[i] == nums[i+1] {
                nums.removeAtIndex(i)
            }
        }
    }

    return nums.count
}

File: ruby/remove_duplicates_from_sorted_array.rb
def remove_duplicates(nums)
  if nums.count > 0
    nums.each_with_index do |num, index|
      while index + 1 < nums.count and num == nums[index+1]
        nums.delete_at(index)
      end
    end
  end
  return nums.count
end
